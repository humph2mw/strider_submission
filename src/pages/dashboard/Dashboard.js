import React, { useState } from 'react';
import { Col, Container, Row } from 'react-bootstrap';
import './style.css';
import jsonData from './receipts.json';
import HighlightCard from '../../components/highlightCard';

export default function Dashboard() {

  // Importing the data from a local json file. Normally,
  // this page would make a fetch request to an api to retrieve
  // the data that it needs. The response would then be stored in 
  // the state variable.
  const [data,] = useState(jsonData);

  return (
    <Container id="dashboardContainer">
      <h1 id="dashboardTitle">Dashboard</h1>
      <Row>
        <Col md={3}>
          <HighlightCard />
        </Col>
        <Col md={3}>
          <HighlightCard />
        </Col>
        <Col md={3}>
          <HighlightCard />
        </Col>
        <Col md={3}>
          <HighlightCard />
        </Col>
      </Row>
    </Container>
  )
}
