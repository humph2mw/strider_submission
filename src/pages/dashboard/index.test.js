import React from 'react'
import TestRenderer from 'react-test-renderer'
import Dashboard from './index'
import { render, screen } from '@testing-library/react'

describe('dashboard tests', () => {
  describe('make sure content is on the page', () => {
    test('should have the correct values for the highlightCards', async () => {
      render(<Dashboard />)

      const highlightCardTitleOne = await screen.findByText('$32')
      const highlightCardDescriptionOne = await screen.findByText('Average Value Per Order')
      const highlightCardTitleTwo = await screen.findByText('2.375')
      const highlightCardDescriptionTwo = await screen.findByText('Average Number of Items Per Order')
      const hightlightCardTitleThree = await screen.findByText('2')
      const highlightCardDescriptionThree = await screen.findByText('Number of Recurring Customers')

      expect(highlightCardTitleOne).toBeTruthy()
      expect(highlightCardTitleOne.textContent).toEqual('$32')
      expect(highlightCardDescriptionOne).toBeTruthy()
      expect(highlightCardDescriptionOne.textContent).toEqual('Average Value Per Order')
      expect(highlightCardTitleTwo).toBeTruthy()
      expect(highlightCardTitleTwo.textContent).toEqual('2.375')
      expect(highlightCardDescriptionTwo).toBeTruthy()
      expect(highlightCardDescriptionTwo.textContent).toEqual('Average Number of Items Per Order')
      expect(hightlightCardTitleThree).toBeTruthy()
      expect(hightlightCardTitleThree.textContent).toEqual('2')
      expect(highlightCardDescriptionThree).toBeTruthy()
      expect(highlightCardDescriptionThree.textContent).toEqual('Number of Recurring Customers')
    })
  })

  describe('snapshot', () => {
    it('renders correctly', () => {
      const tree = TestRenderer
        .create(<Dashboard />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
