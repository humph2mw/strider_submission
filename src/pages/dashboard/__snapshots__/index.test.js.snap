// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard tests snapshot renders correctly 1`] = `
<div
  className="container"
  id="dashboardContainer"
>
  <h1
    id="dashboardTitle"
  >
    Dashboard
  </h1>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    >
      <div
        className="card"
        id="highlightCard"
      >
        <div
          className="card-title h5"
        >
          $0
        </div>
        <p
          className="card-text"
        >
          Average Value Per Order
        </p>
      </div>
    </div>
    <div
      className="col-md-4"
    >
      <div
        className="card"
        id="highlightCard"
      >
        <div
          className="card-title h5"
        >
          0
        </div>
        <p
          className="card-text"
        >
          Average Number of Items Per Order
        </p>
      </div>
    </div>
    <div
      className="col-md-4"
    >
      <div
        className="card"
        id="highlightCard"
      >
        <div
          className="card-title h5"
        >
          0
        </div>
        <p
          className="card-text"
        >
          Number of Recurring Customers
        </p>
      </div>
    </div>
  </div>
  <div
    className="row"
    id="graphRow"
  >
    <div
      className="col-md-6"
    >
      <div
        className="card"
        id="ordersPerDayCard"
      >
        <div
          className="card-title h5"
        >
          Orders Per Day
        </div>
      </div>
    </div>
    <div
      className="col-md-6"
    >
      <div
        className="card"
        id="itemPurchaseFrequencyCard"
      >
        <div
          className="card-title h5"
        >
          Item Order Frequency
        </div>
      </div>
    </div>
  </div>
</div>
`;
